src/bp/core/routers/admin/index.ts

```diff
@@ -129,5 +135,40 @@ export class AdminRouter extends CustomRouter {
     // This way admins could use these routes to push / pull independently of other workspaces.
     // For now we're restricting to super-admin.
     router.use('/versioning', this.checkTokenHeader, assertSuperAdmin, this.versioningRouter.router)
+
+    router.get(
+      '/logs',
+      this.checkTokenHeader,
+      this.needPermissions('read', 'admin.logs'),
+      this.asyncMiddleware(async (req, res) => {
+        const { fromDate, toDate, onlyWorkspace } = req.query
+
+        if (!fromDate || !toDate) {
+          return res.status(400).send('fromDate and toDate must be specified')
+        }
+
+        const from = moment(parseInt(fromDate || ''))
+        const to = moment(parseInt(toDate || ''))
+
+        if (!from.isValid() || !to.isValid()) {
+          return res.status(400).send('fromDate and toDate must be a valid unix timestamp')
+        }
+
+        let botIds
+        if (!req.tokenUser?.isSuperAdmin || yn(onlyWorkspace)) {
+          const botsRefs = await this.workspaceService.getBotRefs(req.workspace)
+          botIds = (await this.botService.findBotsByIds(botsRefs)).filter(Boolean).map(x => x.id)
+        }
+
+        const results = await this.logsRepository.searchLogs({
+          fromDate: from.toDate(),
+          toDate: to.toDate(),
+          botIds,
+          count: req.tokenUser?.isSuperAdmin ? 2000 : 400
```

Comment:
That's a little weird, why do you change the count depending on the user type ?

Reply:
A super admin can view all logs (global + bots of all workspaces) while other users can only see the bots of the currently selected workspace. So I picked a different limit for them, open to discussion

Reply:
So what you say is that because the super admin has access to more stuff, more stuff will show up ... In this case, the admin will simply use the filters properly.

Reply:
Well 400 is the log limit we display in the studio for a bot. I'll remove the difference for consistency then